src/App.tsx

```diff
@@ -1,61 +1,140 @@
+  const [userId, setUserId] = useState(0);
+
+  const [
+    titleErrorMessage,
+    setTitleErrorMessage,
+  ] = useState('');
+  const [
+    userErrorMessage,
+    setUserErrorMessage,
+  ] = useState('');
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title || !userId) {
+      setTitleErrorMessage(!title ? 'Please enter a title' : '');
+      setUserErrorMessage(!userId ? 'Please choose a user' : '');
+
+      return;
+    }
+
+    setTodos(prev => {
+      const newTodo = {
+        id: getNewId(prev),
+        title,
+        completed: false,
+        user: getUserById(userId),
+      };
+
+      return [...prev, newTodo];
+    });
+
+    reset();
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitleErrorMessage('');
+    setTitle(event.target.value);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserErrorMessage('');
+    setUserId(+event.target.value);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/users" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">
+            {'Title: '}
+          </label>
+
+          <input
+            id="titleInput"
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={(event) => handleInputChange(event)}
+          />
+
+          <label
+            htmlFor="titleInput"
+            className="error"
+          >
+            {titleErrorMessage}
+          </label>
+
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label htmlFor="userInput" className="">User: </label>
+
+          <select
+            id="userInput"
+            data-cy="userSelect"
+            value={userId}
+            onChange={handleSelectChange}
+          >
+            <option value={0} disabled>Choose a user</option>
+            {
+              usersFromServer.map(userFromServer => (
+                <option value={userFromServer.id}>{userFromServer.name}</option>
+              ))
+            }
           </select>
 
-          <span className="error">Please choose a user</span>
+          <label
+            htmlFor="userInput"
+            className="error"
+          >
+            {userErrorMessage}
+          </label>
+
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={handleSubmit}
+        >
           Add
         </button>
```

Comment:
Checklist p.6 - While handling form values, trigger submit logic using onSubmit prop of form, instead of onClick submit button